syntax  = "proto3";

package filetransfer;

service FileServer {
	rpc getSlice(SliceRequest) returns  (Slice) {}
	rpc getMetaData(FileName) returns (FileMetadata) {}
}

message SliceRequest {
	int64 idx = 1;
	int64 offset = 2;
}

message Slice {
	int64 idx = 1;
	int64 offset = 2;
	int64 length = 3;
	bytes data = 4;
}

message FileMetadata{
	int64 blockCount = 1;
	int64 blockSize = 2;
	int64 fileSize = 3;
}

message FileName {
}